plugins/woocommerce/src/Admin/Features/ProductBlockEditor/Init.php

```diff
@@ -210,35 +211,136 @@ public function set_current_screen_to_block_editor_if_wc_admin() {
 	}
 
+	/**
+	 * Get default product templates.
+	 *
+	 * @return array The default templates.
+	 */
+	private function get_default_product_templates() {
+		$templates   = array();
+		$templates[] = new ProductTemplate(
+			array(
+				'id'                 => 'standard-product-template',
+				'title'              => __( 'Standard product', 'woocommerce' ),
+				'description'        => __( 'A single physical or virtual product, e.g. a t-shirt or an eBook.', 'woocommerce' ),
+				'order'              => 10,
+				'icon'               => 'shipping',
+				'layout_template_id' => 'simple-product',
+				'product_data'       => array(
+					'type' => 'simple',
+				),
+			)
+		);
+		$templates[] = new ProductTemplate(
+			array(
+				'id'                 => 'grouped-product-template',
+				'title'              => __( 'Grouped product', 'woocommerce' ),
+				'description'        => __( 'A set of products that go well together, e.g. camera kit.', 'woocommerce' ),
+				'order'              => 20,
+				'icon'               => 'group',
+				'layout_template_id' => 'simple-product',
+				'product_data'       => array(
+					'type' => 'grouped',
+				),
+			)
+		);
+		$templates[] = new ProductTemplate(
+			array(
+				'id'                 => 'affiliate-product-template',
+				'title'              => __( 'Affiliate product', 'woocommerce' ),
+				'description'        => __( 'A link to a product sold on a different website, e.g. brand collab.', 'woocommerce' ),
+				'order'              => 30,
+				'icon'               => 'link',
+				'layout_template_id' => 'simple-product',
+				'product_data'       => array(
+					'type' => 'external',
+				),
+			)
+		);
+		$templates[] = new ProductTemplate(
+			array(
+				'id'                 => 'variable-product-template',
+				'title'              => __( 'Variable product', 'woocommerce' ),
+				'description'        => __( 'A product with variations like color or size.', 'woocommerce' ),
+				'order'              => 40,
+				'icon'               => null,
+				'layout_template_id' => 'simple-product',
+				'product_data'       => array(
+					'type' => 'variable',
+				),
+			)
+		);
+
+		// Getting the product types registered via the classic editor.
+		// This is not required when registering new product templates.
+		$registered_product_types = wc_get_product_types();
+
+		foreach ( $registered_product_types as $product_type => $title ) {
+			if ( in_array( $product_type, $this->supported_product_types, true ) ) {
+				continue;
+			}
+
+			$templates[] = new ProductTemplate(
+				array(
+					'id'                 => $product_type . '-product-template',
+					'title'              => $title,
+					'description'        => null,
+					'order'              => ( count( $templates ) + 1 ) * 10,
+					'icon'               => null,
+					'layout_template_id' => null,
+					'product_data'       => array(
+						'type' => $product_type,
+					),
+				)
+			);
+		}
```

Comment:
I still don't think this is the correct way to handle this.

Reply:
done -> https://github.com/woocommerce/woocommerce/pull/41823/commits/bda3b1a236a4a41030be5e57674b820315eabe44